spinnman/model/executable_targets.py

```diff
@@ -9,18 +10,34 @@ class ExecutableTargets(object):
         "_targets",
         "_total_processors"]
 
+    __slots__ = (
+        # the targets
+        "_targets",
+
+        # the total number of processors used
+        "_total_processors",
+
+        # the total core subsets
+        "_all_core_subsets",
+
+        # the mapping between binary and executable type
+        "_binary_to_executable_types_map"
+    )
+
     def __init__(self):
         self._targets = dict()
         self._total_processors = 0
         self._all_core_subsets = CoreSubsets()
+        self._binary_to_executable_types_map = defaultdict(set)
 
-    def add_subsets(self, binary, subsets):
+    def add_subsets(self, binary, subsets, executable_type):
```

Comment:
Add a default for `executable_type` (to the most common form in our users' code?) to reduce the possibility of incompatibilities.

Reply:
uses_simulation?

Reply:
to do this, ill need to move the executable_type from FEC to....... spinnMan. not sure if we want to do this or not, as thats a much more substantial change, and not sure if it has context elsewhere than FEC really.

Reply:
So the issue here is that this now relies, though indirectly, on functionality in FEC i.e. the concept of an executable type.  This doesn't really have meaning in SpiNNMan that I can think of, other than for your filtering here.  It seems this part of the functionality might need a bit of a rethink.